src/quo2/components/switchers/base_card/view.cljs

```diff
@@ -0,0 +1,26 @@
+(ns quo2.components.switchers.base-card.view
+  (:require [quo.react-native :as rn]
+            [quo2.components.buttons.button.view :as button]
+            [quo2.components.switchers.base-card.style :as style]))
+
+(defn base-card
+  [{:keys [banner on-press on-close customization-color]} & children]
+  (let [card-ref (atom nil)]
```

Comment:
This is not a valid form-2 Reagent component. You need to initialize the ratom once. The current code will cause the atom to be initialized on every render of `base-card`.

While doing this, it's a good idea to define the `:ref` function once in the initialization, thus avoiding the unnecessary allocation. Just good practice, not that it matters much.

```clojure
(defn base-card
  []
  (let [card-ref (atom nil)
        set-ref  #(reset! card-ref %)]
    (fn [{:keys [banner on-press on-close customization-color]} & children]
      [rn/touchable-opacity
       {:on-press       on-press
        :ref            set-ref
        :active-opacity 1}
       [rn/view {:style (style/base-container customization-color)}
        (when banner
          [rn/image
           {:source (:source banner)
            :style  {:width 160}}])
        [button/button
         {:size       24
          :type       :grey
          :icon       true
          :on-press   on-close
          :background :photo
          :style      style/close-button}
         :i/close]
        (into [rn/view {:style style/thumb-card}] children)]])))
```